django/db/migrations/executor.py

```diff
@@ -190,19 +190,24 @@ def check_replacements(self):
     def detect_soft_applied(self, project_state, migration):
         """
         Tests whether a migration has been implicitly applied - that the
-        tables it would create exist. This is intended only for use
-        on initial migrations (as it only looks for CreateModel).
+        tables or columns it would create exist. This is intended only for use
+        on initial migrations (as it only looks for CreateModel and AddField).
         """
-        # Bail if the migration isn't the first one in its app
-        if [name for app, name in migration.dependencies if app == migration.app_label]:
+        # Bail if it's NOT an initial migration
```

Comment:
Can you move this comment into the `elif` branch and the one you removed into the `if any()`, please.